inst/cpp/sim.cpp

```diff
@@ -81,7 +92,10 @@ List sim_wrapper_cpp (NumericVector A, List design, List par, NumericVector iov_
   // call ode() once to pre-calculate any initial variables
   // insert A dAdt state_init
   set_covariates(0);
-  prv_dose = doses[0];
+
+  // set prv_dose to first non-zero dose
+  int idx = find_first_nonzero(doses);
```

Comment:
with this change, we are building in a rather quiet assumption that this variable is only correct if doses start at t = 0.

maybe for now this is okay and we can rewrite the model code to something like

```C
F1_avg = F1 \
if (t > 0) { F1_avg = F1 * pow(prv_dose/1000.0, -0.15) } \
```

Reply:
On the other hand, other models also could use prv_dose in the ode_code block, and it is maybe also unexpected for it to be 0 momentarily.

Reply:
yes, my thinking was that if we would keep PKPDsim as-is, and have to make some kind of modification in each model, that would be more annoying, and we might forget for future models. I was thinking this modification, but it's similar to yours.
```
FACT = 1
if(prv_dose > 0) FACT = pow(prv_dose/1000.0, -0.15)
F1_avg = F1 * FACT
```
Neither solution is perfect. Let me look into one other solution I was thinking about though, i.e. we could also make sure that `doses[0]` is not zero to begin with (at least as long as at t=0 there is a dose).